src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -345,42 +347,53 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
           }.toSet
 
           if (allTheRefs.contains(UserService.allUsersGroupRef)) {
-            Future.successful(RequestComplete( ErrorReport(StatusCodes.BadRequest, s"Please contact an administrator to alter access to ${UserService.allUsersGroupRef.groupName}") ))
+            Future.successful(RequestComplete(ErrorReport(StatusCodes.BadRequest, s"Please contact an administrator to alter access to ${UserService.allUsersGroupRef.groupName}")))
+
+          } else if (!updateMap.get(Left(RawlsUser(userInfo))).forall(_ == containerDAO.authDAO.getMaximumAccessLevel(RawlsUser(userInfo), workspaceContext.workspace.workspaceId, txn))) {
```

Comment:
This condition is confusing.  Could you break it up or add comments to make it more clear what's going on?

My assumption is that a check for the existence of the current user in the updateMap would be all that's needed here.